openfl-tutorials/interactive_api/Pytorch_Market_Re-ID/envoy_folder/market_shard_descriptor.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Market shard descriptor."""
+
+import re
+from logging import getLogger
+from pathlib import Path
+
+import numpy as np
+from PIL import Image
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+
+logger = getLogger(__name__)
+
+# Previously download data and put to project folder
+# URL: https://www.kaggle.com/pengcw1/market-1501
+
+# search in whole project directory
+DATAPATH = list(Path.cwd().parents[2].rglob('**/Market'))[0]    # parent directory of project
+
+
+class MarketShardDescriptor(ShardDescriptor):
+    """
+    Market1501 Shard descriptor class.
+
+    Reference:
+    Zheng et al. Scalable Person Re-identification: A Benchmark. ICCV 2015.
+    URL: http://www.liangzheng.org/Project/project_reid.html
+
+    Dataset statistics:
+        identities: 1501 (+1 for background)
+        images: 12936 (train) + 3368 (query) + 15913 (gallery)
+    """
+
+    def __init__(self, rank_worldsize: str = '1,1') -> None:
+        """Initialize MarketShardDescriptor."""
+        super().__init__()
+
+        # Settings for sharding the dataset
+        self.rank, self.worldsize = tuple(int(num) for num in rank_worldsize.split(','))
+
+        self.pattern = re.compile(r'([-\d]+)_c(\d)')
+        self.dataset_dir = Path(DATAPATH)
+        self.train_dir = self.dataset_dir / 'bounding_box_train'
+        self.query_dir = self.dataset_dir / 'query'
+        self.gal_dir = self.dataset_dir / 'bounding_box_test'
+        self.imgs_path = list(self.train_dir.glob('*.jpg'))[self.rank - 1::self.worldsize]
+
+        self._check_before_run()
+
+        self.train, self.num_train_pids, self.num_train_imgs = self._process_dir(self.train_dir)
```

Comment:
`train` list actually contains local paths to data samples which is not safe.
We can leave `query` and `gallery` as is, but try to fill train data loader using `Shard Descriptor's` __getitem__

Reply:
fixed, check please